internal/pkg/deploy/cloudformation/stack/app.go

```diff
@@ -1,157 +1,243 @@
+	return mergeAndFlattenTags(c.AdditionalTags, map[string]string{
+		AppTagKey: c.Project,
 	})
 }
 
-type templateConfigurer interface {
-	Parameters() []*cloudformation.Parameter
-	Tags() []*cloudformation.Tag
+// StackName returns the name of the CloudFormation stack (based on the application name).
+func (c *AppStackConfig) StackName() string {
+	return fmt.Sprintf("%s-infrastructure-roles", c.Project)
 }
 
-func (a *app) templateConfiguration(tc templateConfigurer) (string, error) {
-	doc, err := a.parser.Parse(appParamsTemplatePath, struct {
-		Parameters []*cloudformation.Parameter
-		Tags       []*cloudformation.Tag
-	}{
-		Parameters: tc.Parameters(),
-		Tags:       tc.Tags(),
-	}, template.WithFuncs(map[string]interface{}{
-		"inc": func(i int) int { return i + 1 },
-	}))
-	if err != nil {
-		return "", err
+// StackSetName returns the name of the CloudFormation StackSet (based on the application name).
+func (c *AppStackConfig) StackSetName() string {
+	return fmt.Sprintf("%s-infrastructure", c.Project)
+}
+
+// StackSetDescription returns the description of the StackSet for application resources.
+func (c *AppStackConfig) StackSetDescription() string {
+	return "ECS CLI Application Resources (ECR repos, KMS keys, S3 buckets)"
+}
+
+func (c *AppStackConfig) stackSetAdminRoleName() string {
+	return fmt.Sprintf("%s-adminrole", c.Project)
+}
+
+// StackSetAdminRoleARN returns the role ARN of the role used to administer the Application
+// StackSet.
+func (c *AppStackConfig) StackSetAdminRoleARN() string {
+	//TODO find a partition-neutral way to construct this ARN
+	return fmt.Sprintf("arn:aws:iam::%s:role/%s", c.AccountID, c.stackSetAdminRoleName())
+}
+
+// StackSetExecutionRoleName returns the role name of the role used to actually create
+// Application resources.
+func (c *AppStackConfig) StackSetExecutionRoleName() string {
+	return fmt.Sprintf("%s-executionrole", c.Project)
+}
+
+func (c *AppStackConfig) dnsDelegationAccounts() []string {
+	accounts := append(c.CreateProjectInput.DNSDelegationAccounts, c.CreateProjectInput.AccountID)
+	accountIDs := make(map[string]bool)
+	var uniqueAccountIDs []string
+	for _, entry := range accounts {
+		if _, value := accountIDs[entry]; !value {
+			accountIDs[entry] = true
+			uniqueAccountIDs = append(uniqueAccountIDs, entry)
+		}
 	}
-	return doc.String(), nil
+	return uniqueAccountIDs
 }
 
-func (a *app) addonsOutputs() (*template.ServiceNestedStackOpts, error) {
-	stack, err := a.addons.Template()
-	if err != nil {
-		var noAddonsErr *addons.ErrDirNotExist
-		if !errors.As(err, &noAddonsErr) {
-			return nil, fmt.Errorf("generate addons template for application %s: %w", a.name, err)
+// ToAppRegionalResources takes an Application Resource Stack Instance stack, reads the output resources
+// and returns a modeled  ProjectRegionalResources.
+func ToAppRegionalResources(stack *cloudformation.Stack) (*archer.ProjectRegionalResources, error) {
```

Comment:
Do you think it's possible to move the `ProjectRegionalResources` out of archer into this package?
If it's too much work we can take it up while deprecating the archer pkg

Reply:
That's too much work. And maybe would be better to put it in `config` pkg.